crates/script/src/transaction.rs

```diff
@@ -1,86 +1,41 @@
 use super::ScriptResult;
 use alloy_dyn_abi::JsonAbiExt;
-use alloy_primitives::{hex, Address, Bytes, TxKind, B256};
+use alloy_primitives::{hex, Address, TxKind, B256};
 use eyre::{Result, WrapErr};
+use forge_script_sequence::TransactionWithMetadata;
 use foundry_common::{fmt::format_token_raw, ContractData, TransactionMaybeSigned, SELECTOR_LEN};
 use foundry_evm::{constants::DEFAULT_CREATE2_DEPLOYER, traces::CallTraceDecoder};
 use itertools::Itertools;
 use revm_inspectors::tracing::types::CallKind;
 use serde::{Deserialize, Serialize};
 use std::collections::BTreeMap;
 
-#[derive(Clone, Debug, Default, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct AdditionalContract {
-    #[serde(rename = "transactionType")]
-    pub opcode: CallKind,
-    pub address: Address,
-    pub init_code: Bytes,
-}
-
 #[derive(Clone, Debug, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct TransactionWithMetadata {
-    pub hash: Option<B256>,
-    #[serde(rename = "transactionType")]
-    pub opcode: CallKind,
-    #[serde(default = "default_string")]
-    pub contract_name: Option<String>,
-    #[serde(default = "default_address")]
-    pub contract_address: Option<Address>,
-    #[serde(default = "default_string")]
-    pub function: Option<String>,
-    #[serde(default = "default_vec_of_strings")]
-    pub arguments: Option<Vec<String>>,
-    #[serde(skip)]
-    pub rpc: String,
-    pub transaction: TransactionMaybeSigned,
-    pub additional_contracts: Vec<AdditionalContract>,
-    pub is_fixed_gas_limit: bool,
-}
-
-fn default_string() -> Option<String> {
-    Some(String::new())
-}
-
-fn default_address() -> Option<Address> {
-    Some(Address::ZERO)
-}
-
-fn default_vec_of_strings() -> Option<Vec<String>> {
-    Some(vec![])
+pub struct TxWithMetadata {
+    inner: TransactionWithMetadata,
 }
 
-impl TransactionWithMetadata {
-    pub fn from_tx_request(transaction: TransactionMaybeSigned) -> Self {
-        Self {
-            transaction,
-            hash: Default::default(),
-            opcode: Default::default(),
-            contract_name: Default::default(),
-            contract_address: Default::default(),
-            function: Default::default(),
-            arguments: Default::default(),
-            is_fixed_gas_limit: Default::default(),
-            additional_contracts: Default::default(),
-            rpc: Default::default(),
-        }
+impl From<TxWithMetadata> for TransactionWithMetadata {
```

Comment:
could we avoid having a separate type for this?

Reply:
we can either try keeping all of those building methods on `TransactionWithMetdata`, or do a `ScriptTransactionBuilder` or something, so that eventually we still deal with the same type everywhere